corehq/motech/repeaters/tasks.py

```diff
@@ -71,19 +71,18 @@ def process_repeat_record(repeat_record):
     if repeat_record.cancelled:
         return
 
-    try:
-        repeat_record.repeater
-    except ResourceNotFound:
+    repeater = repeat_record.repeater
+    if not repeater:
         repeat_record.cancel()
         repeat_record.save()
 
     try:
-        if repeat_record.repeater and repeat_record.repeater.paused:
+        if repeater and repeater.paused:
             # postpone repeat record by 1 hour so that these don't get picked in each cycle and
             # thus clogging the queue with repeat records with paused repeater
             repeat_record.postpone_by(timedelta(hours=1))
             return
-        if repeat_record.repeater.doc_type.endswith(DELETED_SUFFIX):
+        if repeater.doc_type.endswith(DELETED_SUFFIX):
```

Comment:
should this check for `repeater` as well?

Reply:
You could `return` after `repeat_record.save()` on line 77. Then you could drop the `repeater and` check on line 80.

Reply:
:+1: 519df5a0b6e